legal-api/src/legal_api/models/amalgamation.py

```diff
@@ -97,10 +97,10 @@ def get_revision_by_id(cls, transaction_id, amalgamation_id):
     def get_revision(cls, transaction_id, business_id):
         """Get amalgamation for the given transaction id."""
         # pylint: disable=singleton-comparison;
-        amalgamation_version = version_class(Amalgamation)
+        amalgamation_version = VersioningProxy.version_class(db.session(), Amalgamation)
         amalgamation = db.session.query(amalgamation_version) \
             .filter(amalgamation_version.transaction_id <= transaction_id) \
-            .filter(amalgamation_version.operation_type == 0) \
+            .filter(amalgamation_version.operation_type == 1) \
```

Comment:
Common Values for operation_type:
INSERT: 1, UPDATE: 2, DELETE: 3

But, looks like the old_version used 0, 1, 2.

Reply:
We should still be using the old db versioning approach of `INSERT: 0, UPDATE: 1, DELETE: 2`.  If this is not the case then the sql_versioning library will need to be updated.

Reply:
I took a look at [sql_versioning's versioning.py](https://github.com/bcgov/lear/blob/4b41a96e4af2420d98694bc79d24641b85251983/python/common/sql-versioning/sql_versioning/versioning.py#L119) and the [insert/update/delete tests](https://github.com/bcgov/lear/blob/4b41a96e4af2420d98694bc79d24641b85251983/python/common/sql-versioning/tests/test_versioning.py) for sql_versioning library.  It looks like the old versioning operation types of `insert: 0, update: 1, delete: 2` are used in the new db versioning.

Are you sure you are seeing `insert: 1, update: 2, delete: 3` being used in the new versioning?

Reply:
I just had the operations(Insert, Update, Delete).
The new_version used `insert: 1, update: 1, delete: 2`
![Uploading Screenshot 2024-10-24 at 12.24.49 PM.png…]()

Reply:
please provide context of where you found this data?  did you create it yourself?  and if you created it yourself, what did you do to get this data?

Reply:
I created the sample endpoint myself for confirmation.


```
@bp.route('/parties', methods=['POST'])
@bp.route('/parties/<int:party_id>', methods=['POST', 'PUT'])
@cross_origin(origin='*')
@jwt.requires_auth
def post_parties(party_id=None):
    print('hello world')
    json_input = copy.deepcopy(request.get_json())
    if party_id:
        party = Party.find_by_id(party_id)
        party.last_name = 'updated_ last name'
    else:
        party = Party(
            first_name='first name',
            last_name='last name',
            delivery_address_id=23718,
            mailing_address_id=23719
        )
    party.save()

    return jsonify(json_input), \
        (HTTPStatus.CREATED if (request.method == 'POST') else HTTPStatus.ACCEPTED)


@bp.route('/parties/<int:party_id>', methods=['DELETE'])
@cross_origin(origin='*')
@jwt.requires_auth
def delete_parties(party_id=None):
    print('hello world')
    json_input = copy.deepcopy(request.get_json())
    if party_id:
        party = Party.find_by_id(party_id)
        party.last_name = 'updated_ last name'
    
        party.delete()
        return jsonify({'message': 'Party has been deleted.'}), HTTPStatus.OK
```

Reply:
Fixed.
Should comment the make_versioned function out.